File: leetcode-submissions/19_Remove_Nth_Node_From_End_of_List.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function removeNthFromEnd(head: ListNode | null, n: number): ListNode | null {
    //now solving this question by using the the two pointer approach that can solve this question in easy way 
    if(head===null || head->next===null){
        return null;
    }
let i:number=0;
let ptr:ListNode=head;

while(i<n){
    if(ptr===null){
        return null;
    }
ptr=ptr.next;
i++;
}
//after that we need to check whether ptr is null or not if null then return head->next
if(ptr===null){
    return head.next;
}
let ptr2:ListNode=head;
while(ptr.next!==null){
    ptr=ptr.next;
    ptr2=ptr2.next;
}
ptr2.next=ptr2.next.next;

};

File: leetcode-submissions/19_Remove_Nth_Node_From_End_of_List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
int getlength(ListNode* head){
    ListNode* temp=head;
    int length=0;
    while(temp!=NULL){
        temp=temp->next;
        length++;
    }
    return length;
}
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        //now we need to find the length of the list
        int length=getlength(head);
        length=length-n;
        int i=1;
        ListNode* temp=head;
        while(i<=length){
            temp=temp->next;
            i++;
        }
        ListNode* temp2=temp->next;
        temp->next=temp->next->next;
        //temp2->next=NULL;
        free(temp2);
        return head;
    }

};